<h3 id="course-material-for"><em>Course material for:</em></h3>
<h1 id="ngs-for-evolutionary-biologists-from-basic-scripting-to-variant-calling">NGS for evolutionary biologists: from basic scripting to variant calling</h1>
<table style="width:100%">
  <tr>
    <td> 
<img src="./Projects/img/elixir_ita_logo.png" alt="yay" height="200" width="200">
</td>
    <td>
Biblioteca, Consiglio Nazionale delle Ricerche <br> via P. Castellino 111<br> 80131 Naples, Italy <br><br><i>May 2-6, 2016</i><br><br><a href="http://bioinformaticstraining.pythonanywhere.com/course/5/" >course announcement</a><br><a href="./prog.md" >course program</a>
</td>
  </tr>
</table>


<p>Recent advances in genome sequencing technologies provide unprecedented opportunities to characterize individual genomes and identify mutations relevant for the study of both evolutionary questions and diagnosis and therapy. This course runs over five days focusing on variant analysis of NGS data and is addressed to wet lab scientists with little or no previous knowledge of the use of a command line, statistical computing packages, and programming. Participants are immersed in the experience of running a complete NGS data analysis work-flow, from the retrieval of raw data from bioinformatics resources, to the downstream analysis of the processed data.</p>
<p>This repository is for the course materials and it is organized as follows:</p>
<table>
<thead>
<tr class="header">
<th align="left">Days</th>
<th align="left">Lessons</th>
<th align="left">Tutorials</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">day 1</td>
<td align="left"><a href="./day1/d1l1_UnixTheory.ppt">Unix/Linux command line</a>, <a href="">Basic R</a> and <a href="./day1/d1l2_NGSPopGen.pdf">NGS and Population Genomics</a></td>
<td align="left"><a href="">Connection to remote server</a>, <a href="./day1/Academis_Linux.pdf">Unix</a> and <a href="">R</a></td>
</tr>
<tr class="even">
<td align="left">day 2</td>
<td align="left"><a href="./day2/d2l1_IntroNGS.pdf">NGS platforms</a> and <a href="./day2/d2l2_DataFormatQC.pdf">Data format an Quality Control</a></td>
<td align="left"><a href="./day2/d2t1_dataQC.html" >Data QC</a></td>
</tr>
<tr class="odd">
<td align="left">day 3</td>
<td align="left"><a href="./day3/d3l1_mapping_BAM_refinement.pdf">Mapping and BAM refinement</a></td>
<td align="left"><a href="d3t1_mapping_and_bam_refinement.md">Mapping</a></td>
</tr>
<tr class="even">
<td align="left">day 4</td>
<td align="left"><a href="./day4/d4l1_SNP_call.pdf">Variant Calling</a></td>
<td align="left"><a href=".day4/d4t1_variantcalling_snps_tutorial.md">SNP calling</a>, <a href="./day4/d4t2_variantcalling_stucturalvariants_tutorial.md">SV calling</a> and <a href="./day4/d4t3_variantcalling_filtering_exercises.md">Filtering</a></td>
</tr>
<tr class="odd">
<td align="left">days 2-5</td>
<td align="left"><a href="./Projects">Practical Projects</a></td>
<td align="left"><a href="Projects/01-Project-01_sum.md">Project 1</a> and <a href="Projects/02-Project-02_sum.md">Project 2</a></td>
</tr>
</tbody>
</table>
